app/src/main/java/com/appunite/loudius/ui/pullrequests/PullRequestsViewModel.kt

```diff
@@ -58,10 +60,26 @@ class PullRequestsViewModel @Inject constructor(
     var state: PullRequestState by mutableStateOf(PullRequestState())
         private set
 
+    private val _isRefreshing = MutableStateFlow(false)
+    val isRefreshing = _isRefreshing.asStateFlow()
```

Comment:
**Question:** What are the advantages of having `isRefreshing` as a separate state flow instead of having `isRefreshing` in `PullRequestState`?

Reply:
Actually, there are a few benefits😅
1. When we use `PullRequestState`, we want to use proper UI (ErrorScreen, LoadingIndicator, PRContent), in this case we have only edited PRContent, not added another screen.
2. In `when` statement we don't have to put the next option witch would be `isRefreshing` and thanks to that the code is more consistent in my opinion, because as I said in the #1, we don't have another screen for that.
3. Whole implementation is much easier and thus code is more readable : -)